src/Grpc/JsonTranscoding/test/Microsoft.AspNetCore.Grpc.JsonTranscoding.Tests/ConverterTests/JsonConverterReadTests.cs

```diff
@@ -20,6 +20,26 @@ public JsonConverterReadTests(ITestOutputHelper output)
         _output = output;
     }
 
+    [Fact]
+    public void NonJsonName()
+    {
+        var json = @"{
+  ""field_name"": ""A field name""
+}";
+
+        AssertReadJson<HelloRequest>(json);
```

Comment:
I dislike this assert, it's basically testing that manual deserialize (with custom options) and our custom JsonParser deserialize create equal output, which doesn't really mean anything since they should be the same thing essentially. We really should be asserting value(s) on HelloRequest are set to the expected value(s).

Reply:
It validates that our JSON functionality matches what is built into Google.Protobuf. The generated objects have an Equals method that does a deep check to validate they're equivalent.

I'll add a couple of manual asserts to sanity check that deserialization is working as expected, but I don't want to assert everything. We know that Google.Protobuf is doing the right thing and if we're equal to it then the code is good.